R/pkg/tests/fulltests/test_sparkSQL.R

```diff
@@ -1803,6 +1803,18 @@ test_that("string operators", {
     collect(select(df4, split_string(df4$a, "\\\\")))[1, 1],
     list(list("a.b@c.d   1", "b"))
   )
+  expect_equal(
+    collect(select(df4, split_string(df4$a, "\\.", 2)))[1, 1],
+    list(list("a", "b@c.d   1\\b"))
+  )
+  expect_equal(
+    collect(select(df4, split_string(df4$a, "b", -2)))[1, 1],
+    list(list("a.", "@c.d   1\\", ""))
+  )
+  expect_equal(
+    collect(select(df4, split_string(df4$a, "b", 0)))[1, 1],
```

Comment:
I wouldn't add all of those cases for R. One test case to check if they can be called should be good enough.

Reply:
per @felixcheung's I added back the `limit = 0` case

Reply:
for context, we've had some cases in the past the wrong value is passed for an parameter - so let's at least get one with and one without any optional parameter

Reply:
@felixcheung just to confirm, things look ok here to you? We now have both with/without optional parameter test cases